### DISCUSSION TITLE
Moving/Renaming pages break all links to that page

### DISCUSSION THREAD

**Discussion by plezops** [requarks/wiki#7412] [Error / Bug Report]
In creating our Wiki and changing pages to be in more logical places, we are breaking lots of referenced links by updating the pages.

To recreate:

1. Create pages: `/page1` and `/page2`
2. Create a link on `/page1` that links to `/page2`
3. Rename `/page2` to `/newdir/page2`

- Expected behavior: the link on `/page1` would update to `/newdir/page2` automatically.
- Actual behavior: will break the link on `/page1` which still points to `/page2` after renaming.

Are there any workarounds or am I doing something wrong?

Seems to be [referenced here](https://requarks.canny.io/wiki/p/moving-a-page-should-update-links) but this hasn't been updated in almost a year and I do not see this as an active issue on Github.

Running 2.5.305
Using /en and /es locales
Node.js 20.18.0
PostgreSQL 11.16
Digital Ocean / Docker Container

**Comment by BPowell76** [comment]
I have also ran into this. The one suggestion that concerns me a bit though is that setting by page idea can cause issues. The best may be to use something like a GUID for referencing then it will be pretty much unique. Using something like that though will cause database bloat if you have a lot of pages.

**Comment by Student4Engineering** [comment]
I also can confirm this

**Comment by gkmzfk6um** [comment]
thats a problem that caused by the content being static wich the expected deafult behavor on the web
- what  can be currently done is [text](link) wich leads to the subpage of said paige even if the page is renamed when its too moved 
- this would be at best a cool feature 👍

**Comment by clonejo** [comment]
Coming from other wiki implementations i expected links to be adapted. The move/rename dialog has no warning.

Redirects from the old to the new path would also be good, you might find that sufficient to cover internal links as well.
